store/src/java/com/zimbra/cs/mailbox/MailSender.java

```diff
@@ -1478,4 +1479,45 @@ public static void unregisterPreSendMailListener(PreSendMailListener listener) {
             }
         }
     }
+
+    /**
+     * provide the Hash for Message-Verification header field.
+     * @param id
+     * @param date
+     * @param from
+     * @throws ServiceException
+     *
+     */
+    private static String getMessageVerificationHeaderValue(String id, Date date, String from) throws MessagingException, ServiceException {
+        String guid = (id + date + from);
+        String guidHash = getHashForMessageVerification(guid, MSGVRFY_ALGORITHM_NAME);
+        String hash = MSGVRFY_HEADER_PREFIX + guidHash;
+        return hash;
+    }
+
+    /**
+     * Create a digest of the given input with the given algorithm.
+     * @param input
+     * @param algo
+     * @throws ServiceException
+     */
+    private static String getHashForMessageVerification(String input, String algo) throws ServiceException {
+        try {
+            MessageDigest md1 = MessageDigest.getInstance(algo);
+            byte[] messageDigest1 = md1.digest(input.getBytes(StandardCharsets.UTF_8));
+            BigInteger number = new BigInteger(1, messageDigest1);
```

Comment:
Does `1` necessary in each variable name?
```suggestion
            MessageDigest md = MessageDigest.getInstance(algo);
            byte[] messageDigest = md.digest(input.getBytes(StandardCharsets.UTF_8));
            BigInteger number = new BigInteger(1, messageDigest);
```